server/src/main/java/org/opensearch/index/remote/RemoteTranslogTransferTracker.java

```diff
@@ -0,0 +1,734 @@
+        }
+
+        synchronized (downloadBytesPerSecMutex) {
+            this.downloadBytesPerSecMovingAverageReference.set(
+                this.downloadBytesPerSecMovingAverageReference.get().copyWithSize(updatedSize)
+            );
+        }
+
+        synchronized (downloadTimeMsMutex) {
+            this.downloadTimeMsMovingAverageReference.set(this.downloadTimeMsMovingAverageReference.get().copyWithSize(updatedSize));
+        }
+    }
+
+    public long getTotalDownloadsSucceeded() {
+        return totalDownloadsSucceeded.get();
+    }
+
+    public void addDownloadsSucceeded(long count) {
+        totalDownloadsSucceeded.addAndGet(count);
+    }
+
+    public long getDownloadBytesSucceeded() {
+        return downloadBytesSucceeded.get();
+    }
+
+    public void addDownloadBytesSucceeded(long count) {
+        downloadBytesSucceeded.addAndGet(count);
+    }
+
+    public long getTotalDownloadTimeInMillis() {
+        return totalDownloadTimeInMillis.get();
+    }
+
+    public void addDownloadTimeInMillis(long duration) {
+        totalDownloadTimeInMillis.addAndGet(duration);
+    }
+
+    public long getLastSuccessfulDownloadTimestamp() {
+        return lastSuccessfulDownloadTimestamp.get();
+    }
+
+    public void setLastSuccessfulDownloadTimestamp(long lastSuccessfulDownloadTimestamp) {
+        this.lastSuccessfulDownloadTimestamp.set(lastSuccessfulDownloadTimestamp);
+    }
+
+    boolean isDownloadBytesMovingAverageReady() {
+        return downloadBytesMovingAverageReference.get().isReady();
+    }
+
+    public double getDownloadBytesMovingAverage() {
+        return downloadBytesMovingAverageReference.get().getAverage();
+    }
+
+    public void updateDownloadBytesMovingAverage(long count) {
+        synchronized (downloadBytesMutex) {
+            this.downloadBytesMovingAverageReference.get().record(count);
+        }
+    }
+
+    boolean isDownloadBytesPerSecMovingAverageReady() {
+        return downloadBytesPerSecMovingAverageReference.get().isReady();
+    }
+
+    public double getDownloadBytesPerSecMovingAverage() {
+        return downloadBytesPerSecMovingAverageReference.get().getAverage();
+    }
+
+    public void updateDownloadBytesPerSecMovingAverage(long speed) {
+        synchronized (downloadBytesPerSecMutex) {
+            this.downloadBytesPerSecMovingAverageReference.get().record(speed);
+        }
+    }
+
+    boolean isDownloadTimeMovingAverageReady() {
+        return downloadTimeMsMovingAverageReference.get().isReady();
+    }
+
+    public double getDownloadTimeMovingAverage() {
+        return downloadTimeMsMovingAverageReference.get().getAverage();
+    }
+
+    public void updateDownloadTimeMovingAverage(long duration) {
+        synchronized (downloadTimeMsMutex) {
+            this.downloadTimeMsMovingAverageReference.get().record(duration);
+        }
+    }
+
+    public void recordDownloadStats(long bytesBefore, long downloadStartTime) {
```

Comment:
pls add java doc here and other places where the name is not enough to understand the behaviour of the method.

Reply:
you could probably move the comment download below to the java doc and add a bit more context.